azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.rampup;
+
+import azkaban.imagemgmt.daos.ImageRampupDao;
+import azkaban.imagemgmt.daos.ImageTypeDao;
+import azkaban.imagemgmt.daos.ImageVersionDao;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.models.ImageRampup;
+import azkaban.imagemgmt.models.ImageType;
+import azkaban.imagemgmt.models.ImageVersion;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for fetching the version of the available images based on currently
+ * active rampup plan and rampup details or the version which is already ramped up and active. Here
+ * is the version selection process for an image type - 1. Fetch the rampup details for the given
```

Comment:
Nit: New line for all the numbering

Reply:
Tried multiple times earlier. But formatting is changing that.